multiuser/api/che-multiuser-api-organization/src/main/java/org/eclipse/che/multiuser/organization/api/permissions/OrganizationRemoteSubscriptionPermissionsChecks.java

```diff
@@ -0,0 +1,110 @@
+import org.eclipse.che.multiuser.api.permission.server.jsonrpc.RemoteSubscriptionPermissionManager;
+import org.eclipse.che.multiuser.api.permission.server.model.impl.AbstractPermissions;
+import org.eclipse.che.multiuser.organization.api.listener.OrganizationEventsWebsocketBroadcaster;
+
+/**
+ * Holds and registers permissions checks for organization related events.
+ *
+ * <p>Covers events published via {@link OrganizationEventsWebsocketBroadcaster}.
+ *
+ * @author Sergii Leshchenko
+ */
+@Singleton
+public class OrganizationRemoteSubscriptionPermissionsChecks {
+
+  private final PermissionsManager permissionsManager;
+
+  @Inject
+  public OrganizationRemoteSubscriptionPermissionsChecks(PermissionsManager permissionsManager) {
+    this.permissionsManager = permissionsManager;
+  }
+
+  @Inject
+  public void register(RemoteSubscriptionPermissionManager permissionFilter) {
+    OrganizationMembershipsChangedSubscriptionPermissionsCheck membershipsEventsCheck =
+        new OrganizationMembershipsChangedSubscriptionPermissionsCheck();
+
+    permissionFilter.registerCheck(membershipsEventsCheck, ORGANIZATION_MEMBERSHIP_METHOD_NAME);
+
+    OrganizationChangedSubscriptionPermissionsCheck organizationChangedCheck =
+        new OrganizationChangedSubscriptionPermissionsCheck(permissionsManager);
+    permissionFilter.registerCheck(organizationChangedCheck, ORGANIZATION_CHANGED_METHOD_NAME);
+  }
+
+  @VisibleForTesting
+  static class OrganizationMembershipsChangedSubscriptionPermissionsCheck
+      implements RemoteSubscriptionPermissionCheck {
+
+    @Override
+    public void check(String methodName, Map<String, String> scope) throws ForbiddenException {
+      String userId = scope.get("userId");
+      if (userId == null) {
+        throw new ForbiddenException("User id must be specified in scope");
+      }
+
+      String currentUserId = EnvironmentContext.getCurrent().getSubject().getUserId();
+
+      if (!currentUserId.equals(userId)) {
+        throw new ForbiddenException("It is only allowed to listen to own memberships changes");
+      }
+    }
+  }
+
+  @VisibleForTesting
+  static class OrganizationChangedSubscriptionPermissionsCheck
+      implements RemoteSubscriptionPermissionCheck {
+
+    private final PermissionsManager permissionsManager;
+
+    public OrganizationChangedSubscriptionPermissionsCheck(PermissionsManager permissionsManager) {
+      this.permissionsManager = permissionsManager;
+    }
+
+    @Override
+    public void check(String methodName, Map<String, String> scope) throws ForbiddenException {
+      String organizationId = scope.get("organizationId");
+      if (organizationId == null) {
+        throw new ForbiddenException("Organization id must be specified in scope");
+      }
+
+      String currentUserId = EnvironmentContext.getCurrent().getSubject().getUserId();
+
+      try {
+        AbstractPermissions permissions =
```

Comment:
Can you add a comment to the code with the explanation of why this part is implemented in this way?